src/Node.php

```diff
@@ -374,7 +380,7 @@ public function & getFileContents() : string {
      * @return Node|null
      */
     public function getDescendantNodeAtPosition(int $pos) {
-        $descendants = iterator_to_array($this->getDescendantNodes());
+        $descendants = iterator_to_array($this->getDescendantNodes(), false);
```

Comment:
Looks like iterator_to_array is unnecessarily getting used in a bunch of places, when iterating through elements of the generator should suffice. Just filed #39 to track this.